lib/CodeBuilder/Tests/Adapter/UpdaterTestCase.php

```diff
@@ -1545,6 +1596,33 @@ public function hello(
                     EOT
             ];
 
+        yield 'It modifies parameter types' => [
+                <<<'EOT'
+                    class Foo {
+                        public function changeMe(OldClass $dependency)
+                        {
+                        }
+                    }
+                    EOT,
+                SourceCodeBuilder::create()
+                    ->class('Foo')
+                        ->method('changeMe')
+                            ->parameter('dependency')
+                                ->type('SomeOtherClass')
+                            ->end()
+                        ->end()
+                    ->end()
+                    ->build()
+                ,
+                <<<'EOT'
+                    class Foo {
+                        public function changeMe(SomeOtherClass $dependency)
+                        {
+                        }
+                    }
+                    EOT
+        ];
```

Comment:
I have no idea why this test is failing. Should I try to fix it or just leave it be for now?

Reply:
```diff
--- a/lib/CodeBuilder/Adapter/TolerantParser/Updater/AbstractMethodUpdater.php
+++ b/lib/CodeBuilder/Adapter/TolerantParser/Updater/AbstractMethodUpdater.php
@@ -162,9 +162,7 @@ abstract class AbstractMethodUpdater
         // Adding new parameters to the mix
         foreach ($parameters as $parameter) {
             assert($parameter instanceof PhpactorParameter);
-            if (!isset($renderedParameters[$parameter->name()])) {
-                $renderedParameters[$parameter->name()] = $this->renderer->render($parameter>
-            }
+            $renderedParameters[$parameter->name()] = $this->renderer->render($parameter);
         }
 
         $startPosition = $methodDeclaration->openParen->getStartPosition();
```

makes it pass - it simply was not updating parameters. the question is do we feel confident that it won't mess up existing parameters by unconditionally replacing them?

Reply:
Since phpactor still doesn't know all the ins and outs of constuctor parameters (eg. `readonly` keyword) I wouldn't recommend that. But I have to admit that would also be one of the next things I will be working on.

Reply:
yep, so we don't support modifying parameters currently then, and this test can be removed :+1: